### PR TITLE
[FIX] core: fix search with accent

### PR DESCRIPTION
**Pull request by HydrionBurst**
For a trigram indexed translated field field_x,
domain leaf ('field_x', '=', value) and ('field_x', 'ilike', pattern) will miss records if the pattern contains accent characters

This commit set ensure_ascii=False for json.dumps to avoid escaping special characters

opw-3048182

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by HydrionBurst** [comment]
@rco-odoo 
Find an easy fix
However, since python `json.dumps(pattern, ensure_ascii=False)` may not always return the same result as `::text` for jsonb value in PostgreSQL. If there are more corner cases from customers in the future, we may have to use `to_jsonb(pattern::text)::text` for escaping

**Review by rco-odoo — CHANGES_REQUESTED** [review]
I think that you should add some test to check that `search()` works as expected in that case.  To me this would be useful to ensure that _both_ SQL conditions (the main condition _and_ the one using the index) work as expected.

**Review by xmo-odoo — APPROVED** [review]

File: odoo/addons/base/tests/test_translate.py [hunk 1]
```diff
@@ -885,6 +886,7 @@ def test_pattern_conversion(self):
             (r'a\bc',     '%abc%',     r'redundant \ for pattern should be removed'),
             ( 'abc_de',   '%abc%',      'sub patterns less than 3 characters should be dropped'),
             ( 'ab',       '%',          'patterns without trigram should be simplified'),
+            ('Æéç°²ð',    '%Æéç°²ð%',   'Special characters should not be escaped'),
```
Comment (xmo-odoo) [inline odoo/addons/base/tests/test_translate.py:889]: Might be a good idea to also check very high-range characters e.g. emoji, CJK Compatibility Ideographs Supplement block, ...

Also testing the escaping seems not very interesting as what we're really interested in is *the search*. Maybe that could be tested more extensively with similarly odd search patterns? (with positive and negative matches both)

**Comment by HydrionBurst** [comment]
@rco-odoo Code has been updated and rebased, and the runbot is green. Could you review it again?

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**